akka-docs/rst/scala/remoting-artery.rst

```diff
@@ -0,0 +1,469 @@
+
+In cases, where Network Address Translation (NAT) is used or other network bridging is involved, it is important
+to configure the system so that it understands that there is a different between his externally visible, canonical
+address and between the host-port pair that is used to listen for connections. See :ref:`remote-configuration-nat-artery`
+for details.
+
+Aquiring references to remote actors
+------------------------------------
+
+In order to communicate with an actor, it is necessary to have its :class:`ActorRef`. In the local case it is usually
+the creator of the actor (the caller of ``actorOf()``) is who gets the :class:`ActorRef` for an actor that it can
+then send to other actors. Alternatively, an actor can look up another located at a known path using
+:class:`ActorSelection`. These methods are available even in remoting enabled systems:
+
+* Remote Lookup    : used to look up an actor on a remote node with ``actorSelection(path)``
+* Remote Creation  : used to create an actor on a remote node with ``actorOf(Props(...), actorName)``
+
+In the next sections the two alternatives are described in detail.
+
+
+Looking up Remote Actors
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+``actorSelection(path)`` will obtain an ``ActorSelection`` to an Actor on a remote node, e.g.::
+
+  val selection =
+    context.actorSelection("akka.tcp://actorSystemName@10.0.0.1:2552/user/actorName")
+
+As you can see from the example above the following pattern is used to find an actor on a remote node::
+
+  akka://<actor system>@<hostname>:<port>/<actor path>
+
+.. note::
+
+  Unlike with earlier remoting, the protocol field is always `akka` as pluggable transports are no longer supported.
+
+Once you obtained a selection to the actor you can interact with it in the same way you would with a local actor, e.g.::
+
+  selection ! "Pretty awesome feature"
+
+To acquire an :class:`ActorRef` for an :class:`ActorSelection` you need to
+send a message to the selection and use the ``sender`` reference of the reply from
+the actor. There is a built-in ``Identify`` message that all Actors will understand
+and automatically reply to with a ``ActorIdentity`` message containing the
+:class:`ActorRef`. This can also be done with the ``resolveOne`` method of
+the :class:`ActorSelection`, which returns a ``Future`` of the matching
+:class:`ActorRef`.
+
+.. note::
+
+  For more details on how actor addresses and paths are formed and used, please refer to :ref:`addressing`.
+
+.. note::
+
+  Message sends to actors that are actually in the sending actor system do not
+  get delivered via the remote actor ref provider. They're delivered directly,
+  by the local actor ref provider.
+
+  Aside from providing better performance, this also means that if the hostname
+  you configure remoting to listen as cannot actually be resolved from within
+  the very same actor system, such messages will (perhaps counterintuitively)
+  be delivered just fine.
```

Comment:
A bit weird with two notes after each other, maybe the first one should just be a regular paragraph?